### ISSUE TITLE
False positive for no-extra-semicolons in CSS-in-JS

### ISSUE THREAD

**Issue by hudochenkov** [stylelint/stylelint#6563]
### What steps are needed to reproduce the bug?

Originally reported in https://github.com/hudochenkov/postcss-styled-syntax/issues/1

CSS-in-JS using [postcss-styled-syntax](https://github.com/hudochenkov/postcss-styled-syntax) as custom syntax:

```tsx
const RedBox = styled.div`
  ${css`
    background-color: red;
  `}
`;
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": [
    "stylelint-config-recommended",
    "stylelint-config-styled-components"
  ],
  "customSyntax": "postcss-styled-syntax"
}
```

### How did you run Stylelint?

CLI, API, doesn't matter.

### Which version of Stylelint are you using?

`14.16.0`

### What did you expect to happen?

No violation is triggered.

### What actually happened?

Semicolon after `red` triggers "Unexpected extra semicolon" (no-extra-semicolons) violation.

### Does the bug relate to non-standard syntax?

CSS-in-JS with postcss-styled-syntax

### Proposal to fix the bug

This false positive appears because `postcss-styled-syntax` puts interpolations `${}` in `raws.before` and `raws.after`. Interpolations inside `raws` are not meant to be analyzed. Because content of interpolation (in example above it's `background-color: red;`) is added as a separate `Root` node and analyzed by stylelint correctly. `no-extra-semicolons` rule doesn't know about this, so it analyzes `raws` and reports a violation.

To fix this we would need to change the rule specifically for `postcss-styled-syntax` to avoid checking interpolation bit of `raws.before` and `raws.after`.

I remember back in the days there was a direction in Stylelint development towards only CSS syntax in rules. However, there are still plenty of non-standard syntax-specific logic in a code base. [Including rule](https://github.com/stylelint/stylelint/blob/5e6d46aad5f61ef86f5823a67696a137a0e77a10/lib/rules/no-extra-semicolons/index.js#L141) in question.

I can't contribute a fix to the rule, but it would mean that there would be a `postcss-styled-syntax`-specific logic. Is it a good direction?

I know that it is always possible to create plugin like [stylelint-scss](https://github.com/stylelint-scss/stylelint-scss), but I would like not to add another package to maintain just yet :)

**Comment by ybiquitous** [comment]
@hudochenkov The `no-extra-semicolons` rule will be deprecated in the next release v15.0.0 and be removed in v16.0.0:

https://github.com/stylelint/stylelint/blob/387456648ed03a9e04f4602d5771080ff8716f55/lib/rules/no-extra-semicolons/README.md?plain=1#L3

https://github.com/stylelint/stylelint/blob/387456648ed03a9e04f4602d5771080ff8716f55/docs/user-guide/rules.md?plain=1#L261

So I think it is nice to just turn off the rule. Is there any problem?

**Comment by hudochenkov** [comment]
This even better! Thank you! :)
